src/state/witness_generator.rs

```diff
@@ -600,7 +609,37 @@ impl WitnessGenerator {
         while i + self.n_tx <= len {
             let block = Self::forge_with_txs(self.block_generate_num, &self.buffered_txs[i..i + self.n_tx]);
             blocks.push(block);
+
```

Comment:
pop_all_blocks is too long.

I suggest split the new codes into a standalone method